iree/compiler/Dialect/HAL/Transforms/test/inline_device_switches.mlir

```diff
@@ -0,0 +1,92 @@
+// RUN: iree-opt -split-input-file -iree-hal-inline-device-switches -canonicalize %s | IreeFileCheck %s
+
+// CHECK-LABEL: @simple_constants
+// CHECK-SAME: %[[DEVICE:.+]]: !hal.device
+// CHECK-SAME: %[[ARG:.+]]: i32
+func @simple_constants(%device : !hal.device, %arg : i32) -> i32 {
+  // CHECK-DAG: %[[C0:.+]] = constant 0
+  %c0 = constant 0 : i32
+  // CHECK-DAG: %[[C1:.+]] = constant 1
+  %c1 = constant 1 : i32
+  // CHECK-DAG: %[[C2:.+]] = constant 2
+  %c2 = constant 2 : i32
+  // CHECK-DAG: %[[C3:.+]] = constant 3
+  // CHECK-DAG: %[[C4:.+]] = constant 4
+  %0 = hal.device.switch(%device : !hal.device) -> i32
+    // CHECK-NEXT: %[[IS0:.+]] = hal.device.match.id %[[DEVICE]], pattern = ["vulkan-v1.?-*"] : (!hal.device) -> i1
+    // CHECK-NEXT: cond_br %[[IS0]], ^bb1, ^bb2
```

Comment:
nope, too painful and not valuable (would then need more ways to explicitly distinguish empty branches, and changes to branch folding would break things)